src/plugins/intel_gpu/tests/test_cases/gather_gpu_test.cpp

```diff
@@ -6,53 +6,837 @@
+                        for(int m=0;m<fsv;m++)
+                            ASSERT_EQ((float)ans[i*f2*y2*x2 + (j*fsv+m)*y2*x2 + k*x2 + l],
+                            (float)float16_to_float32(output_ptr[i*f2/fsv*y2*x2*fsv + j*y2*x2*fsv + k*x2*fsv + l*fsv + m]));
+    }
+    // void TearDown() override {}
+};
+TEST_F(gather8fsv16yFixt, a){}
+TEST_F(gather8fsv16yFixt, b){}
+TEST_F(gather8fsv16yFixt, c){}
+
+class gather8Fixture1 : public ::testing::Test {
+protected:
+    std::vector<FLOAT16> dat;
+    std::vector<float> ind;
+    std::vector<FLOAT16> ans;
+    size_t b0=2, f0=3, y0=4, x0=1;
+    size_t b1=5, f1=6, y1=1, x1=1;
+    size_t b2=2, f2=3, y2=5, x2=6;
+    int axis=2;//y
+    int batch_dim=0;
+
+    void SetUp() override {
+        auto& engine = get_test_engine();
+
+        dat=generate_random_1d<FLOAT16>(2*3*4*1,0,99);
+        auto input0 = engine.allocate_memory({ data_types::f16, format::bfyx,  { b0, f0, x0, y0 } }); // Dictionary
+
+        ind=generate_random_1d<float>(5*6*1*1,0,input0->get_layout().get_dim(axis)-1,1);
+        auto input1 = engine.allocate_memory({ data_types::f32, format::bfyx, { b1, f1, x1, y1 } }); // Indexes
+
+        set_values(input0, dat);
+        set_values(input1, ind);
+
+        topology topology;
+        topology.add(input_layout("InputDictionary", input0->get_layout()));
+        topology.add(input_layout("InputText", input1->get_layout()));
+        topology.add(gather("gather", "InputDictionary", "InputText", axis, {b2,f2,y2,x2}, batch_dim, true, "", cldnn::padding(), format::bfyx));
+
+        network network(engine, topology);
+        network.set_input_data("InputDictionary", input0);
+        network.set_input_data("InputText", input1);
+
+        auto output = network.execute().at("gather").get_memory();
+        cldnn::mem_lock<uint16_t> output_ptr(output, get_test_stream());
+        
+        auto to_vec_size_t=[](const std::vector<int>& vec){return std::vector<size_t>(vec.begin(),vec.end());};
+        auto logical_dim=[](std::vector<int> a){ while(a.size()&&a.back()==1)a.pop_back(); return a.size(); };
+        ans=std::vector<FLOAT16>(2*3*5*6);
+        ngraph::runtime::reference::gather<FLOAT16,float>(
+            dat.data(),
+            ind.data(),
+            ans.data(),
+            ov::Shape(to_vec_size_t(input0->get_layout().get_dims())),
+            ov::Shape(to_vec_size_t(input1->get_layout().get_dims())),
+            ov::Shape(to_vec_size_t(output->get_layout().get_dims())),
+            axis,
+            batch_dim>=0?batch_dim:batch_dim+logical_dim(input1->get_layout().get_dims()));
+        for (size_t i = 0; i < ans.size(); ++i)
+            ASSERT_EQ((float)ans[i], (float)float16_to_float32(output_ptr[i]));
+        
+    }
+    // void TearDown() override {}
+};
+TEST_F(gather8Fixture1, myfixturetest0){}
+TEST_F(gather8Fixture1, myfixturetest1){}
+TEST_F(gather8Fixture1, myfixturetest2){}
+TEST_F(gather8Fixture1, myfixturetest3){}
```

Comment:
For testing various layouts, I think you can use `TEST_P`, instead of duplicating code.

Reply:
Updated to parameterized version.